### ISSUE TITLE
Add history to Find and Replace text boxes, perhaps via dropdown

### ISSUE THREAD

**Issue by aswiseman** [microsoft/vscode#114797]
When you are carrying out a number of find/replace operations on a number of files, it would be handy to have a history of what you found and what you replaced with. This could be simply implemented by having a dropdown of previous values in the Find and Replace boxes and is how my former favourite text editor did it. 

If you wanted to be more sophisticated, you could perhaps suggest the matching "Replace" value when the user chooses a "Find" value from the history.

**Comment by ArturoDent** [comment]
In case you are unaware, up/down arrows do provide the find and replace history - but not synced or via a dropdown as you suggest.

**Comment by aswiseman** [comment]
Thanks for this @ArturoDent - I hadn't realised you could that! 
It may be that some sort of auto-complete would make this more discoverable?

**Comment by IllusionMH** [comment]
Duplicate of #23099 if you skip "synced fields" part

**Comment by aswiseman** [comment]
@IllusionMH - I'm happy to add the 'synced fields' suggestion to #23099 and then close this issue, if people prefer?

**Comment by IllusionMH** [comment]
I think it make sense to track "synced" find/replace history separately from dropdpwn because it brings it own specific challenges which aren't directly related to dropdown.
My comment was about fact that there is already request for dropdown.

@JacksonKearl @roblourens what would be preference here?

**Comment by roblourens** [comment]
If tracking multiple pairs of search/replace is important to you, then you might just want to keep multiple search editors open. This is sort of like what was discussed in https://github.com/microsoft/vscode/issues/16488, not quite though.

I am not really sure how this request would work. I think it's more expected to keep the two histories separate. It probably depends on #23099, because it seems like you would need some extra UX for it.
